src/analysis/normed_space/weak_dual_of_normed_space.lean

```diff
@@ -0,0 +1,163 @@
+
+lemma to_original_dual_injective :
+  function.injective (λ (x' : weak_dual 𝕜 E), x'.to_original_dual) := λ _ _ , id
+
+-- Q: Are the following simp-lemmas useful?
+
+@[simp] lemma to_weak_dual_to_original_dual (x' : weak_dual 𝕜 E) :
+  (x'.to_original_dual).to_weak_dual = x' := rfl
+
+@[simp] lemma to_original_dual_to_weak_dual (x' : dual 𝕜 E) :
+  (x'.to_weak_dual).to_original_dual = x' := rfl
+
+@[simp] lemma to_weak_dual_inj_iff (x' y' : dual 𝕜 E) :
+  x'.to_weak_dual = y'.to_weak_dual ↔ x' = y' := iff.rfl
+
+@[simp] lemma to_original_dual_inj_iff (x' y' : weak_dual 𝕜 E) :
+  x'.to_original_dual = y'.to_original_dual ↔ x' = y' := iff.rfl
+
+/-- The linear equivalence between `dual 𝕜 E` and `weak_dual 𝕜 E` for a normed space `E`. -/
+def linequiv_to_weak_dual : dual 𝕜 E ≃ₗ[𝕜] weak_dual 𝕜 E :=
+{ to_fun := (λ (x' : dual 𝕜 E), x'.to_weak_dual),
+  map_add' := by { intros x' y', refl, },
+  map_smul' := by { intros c x', refl, },
+  inv_fun := (λ (x' : weak_dual 𝕜 E), x'.to_original_dual),
+  left_inv := to_original_dual_to_weak_dual,
+  right_inv := to_weak_dual_to_original_dual, }
+
+@[simp]
+lemma linequiv_to_weak_dual_apply (x' : dual 𝕜 E) :
+  linequiv_to_weak_dual x' = x'.to_weak_dual := rfl
+
+@[simp]
+lemma equiv_to_weak_dual_symm_apply (x' : weak_dual 𝕜 E) :
+  linequiv_to_weak_dual.symm x' = x'.to_original_dual := rfl
+
+-- TODO: The only reason to separate this from `evaluate_dual_at` is to get access to the proofs
+-- of `map_add'` and `map_smul'`. Surely there is a syntax to avoid this unnecessary intermediate
+-- step... right?
+def normed_space.evaluate_dual_at' (z : E) : (dual 𝕜 E) →ₗ[𝕜] 𝕜 :=
+{ to_fun := (λ (x' : dual 𝕜 E), x' z),
+  map_add' := by simp only [forall_const, eq_self_iff_true, continuous_linear_map.add_apply],
+  map_smul' := by simp only [forall_const, eq_self_iff_true, pi.smul_apply,
+                             continuous_linear_map.coe_smul'], }
+
+/-
+TODO: Is there a way to make the following dot notation work?
+(And the same for `evaluate_dual_at`?)
+
+variables (w : E)
+#check w
+#check normed_space.evaluate_dual_at' w
+#check w.evaluate_dual_at' -- fails
+-/
+
+def normed_space.evaluate_dual_at (z : E) : (dual 𝕜 E) →L[𝕜] 𝕜 :=
+{ to_fun := (λ (x' : dual 𝕜 E), x' z),
+  map_add' := (normed_space.evaluate_dual_at' z).map_add,
+  map_smul' := (normed_space.evaluate_dual_at' z).map_smul,
+  cont := begin
+    apply @continuous_of_linear_of_bound 𝕜 (dual 𝕜 E) 𝕜 _ infer_instance _ _ _
+      (λ (x' : dual 𝕜 E), x' z) (normed_space.evaluate_dual_at' z).map_add
+      (normed_space.evaluate_dual_at' z).map_smul (∥ z ∥),
+    intros x',
+    have key := continuous_linear_map.le_op_norm x' z,
+    rwa mul_comm at key,
+  end, }
+
+theorem to_weak_dual_continuous :
+  continuous (λ (x' : dual 𝕜 E), x'.to_weak_dual) :=
+begin
+  apply continuous_induced_rng,
+  apply continuous_pi_iff.mpr,
+  intros z,
+  exact (normed_space.evaluate_dual_at z).continuous,
```

Comment:
```suggestion
  exact (inclusion_in_double_dual 𝕜 E z).continuous,
```
Actually the construction `normed_space.evaluate_dual_at` exists already under this other name.